### PR TITLE
Openqasm local simulator

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Add schemas for the openqasm native simulator for input/output support. Openqasm program schemas extended to take inputs and a new result type added to support output. `OQ3ProgramResult` is an analog of [`GateModelTaskResult`](https://github.com/aws/amazon-braket-schemas-python/blob/main/src/braket/task_result/gate_model_task_result_v1.py#L38).

*Testing done:*

e2e tests in other repos covering these schema changes

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-schemas-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/ir/openqasm/program_v1.py [hunk 1]
```diff
@@ -32,3 +38,9 @@ class Program(BraketSchemaBase):
     _PROGRAM_HEADER = BraketSchemaHeader(name="braket.ir.openqasm.program", version="1")
     braketSchemaHeader: BraketSchemaHeader = Field(default=_PROGRAM_HEADER, const=_PROGRAM_HEADER)
     source: str
+    inputs: Optional[
```
Comment (jheckey) [inline src/braket/ir/openqasm/program_v1.py:41]: Are these related to `input` and `output` keywords in OQ3? What is the expected usage for the values here?

Comment (ajberdy) [reply src/braket/ir/openqasm/program_v1.py:41]: Yes, exactly. The demos in the simulator PR show how these are used

**Review by jheckey — APPROVED** [review]
